Userland/Libraries/LibWeb/Layout/SVGSVGBox.cpp

```diff
@@ -75,9 +75,10 @@ Optional<CSSPixelFraction> SVGSVGBox::calculate_intrinsic_aspect_ratio() const
         auto const& viewbox = dom_node().view_box().value();
 
         // 2. return viewbox.width / viewbox.height
+        auto viewbox_width = CSSPixels::nearest_value_for(viewbox.width);
         auto viewbox_height = CSSPixels::nearest_value_for(viewbox.height);
-        if (viewbox_height != 0)
-            return CSSPixels::nearest_value_for(viewbox.width) / viewbox_height;
+        if (viewbox_width != 0 && viewbox_height != 0)
```

Comment:
Could you file a spec bug for this? It really seems like the 0 cases should be handled explicitly in the algorithm, here and in steps 1.1 and 2.2.